### ISSUE TITLE
Fog of war doesnt work properly after updating to Sonoma 14.0

### ISSUE THREAD

**Issue by CDOIS** [ValveSoftware/Dota2-Gameplay#12227]
### Description

Fog not displaying properly after updating the new apple OS.

### Screenshots

<img width="255" alt="Screenshot 2023-09-27 at 7 08 11 PM" src="https://github.com/ValveSoftware/Dota2-Gameplay/assets/19749418/7e444e39-8960-478e-b721-be9b32cadf73">
<img width="1680" alt="Screenshot 2023-09-27 at 6 59 28 PM" src="https://github.com/ValveSoftware/Dota2-Gameplay/assets/19749418/c210259c-2864-47b2-863a-6700e0399255">
<img width="1680" alt="Screenshot 2023-09-27 at 6 59 25 PM" src="https://github.com/ValveSoftware/Dota2-Gameplay/assets/19749418/b65574f7-9ced-4155-aba1-02d4ed30005f">
<img width="1680" alt="Screenshot 2023-09-27 at 6 59 22 PM" src="https://github.com/ValveSoftware/Dota2-Gameplay/assets/19749418/ce49ee2b-7e03-4d40-a348-8ae36ce0713e">
![Screenshot 2023-09-27 at 6 59 09 PM](https://github.com/ValveSoftware/Dota2-Gameplay/assets/19749418/ad6bef90-6f5b-48a0-8cd9-ec1e1108bbbe)

**Comment by zaczh** [comment]
Any workaround on this issue?

**Comment by TobyFlendersonFromHR** [comment]
yes

**Comment by ToanHoangHuy** [comment]
Same issue. Can anyone help me to fix this, please? I can't play DotA normally. playing dota 2 on Ventura always made me feel so good. But why that doesn't appear on Sonoma.

**Comment by yxwzyyk** [comment]
same question

**Comment by sl476** [comment]
Same quesiton. It's annoying.

**Comment by pototsky-rem** [comment]
Try to change resolution, all kind of resolutions

**Comment by ToanHoangHuy** [comment]
> Try to change resolution, test all kind of resolutions in game This problem was solved for me when I selected the screen resolution 1024x640 (60hz)

I set up the game to windows mode, resolution really changed lower, but... maybe i will leave game if valve doesn't have plan to fix it

**Comment by sl476** [comment]
> > > Try to change resolution, test all kind of resolutions in game This problem was solved for me when I selected the screen resolution 1024x640 (60hz)
> > 
> > 
> > I set up the game to windows mode, resolution really changed lower, but i will try all resolutions for checking, maybe i will leave game if valve doesn't have plan to fix it
> 
> the problem doesn't resolve, good bye dota 2, i will remember you ....

Have you tried ALL of the available resolutions?
Mine seems to have resolved now...

**Comment by ToanHoangHuy** [comment]
> > > > Try to change resolution, test all kind of resolutions in game This problem was solved for me when I selected the screen resolution 1024x640 (60hz)
> > > 
> > > 
> > > I set up the game to windows mode, resolution really changed lower, but i will try all resolutions for checking, maybe i will leave game if valve doesn't have plan to fix it
> > 
> > 
> > the problem doesn't resolve, good bye dota 2, i will remember you ....
> 
> Have you tried ALL of the available resolutions? Mine seems to have resolved now...

What are spec pc or mac you using? Can you record video before and afer applying your changed resolution? I tried all resolutions but no changed anything

**Comment by pototsky-rem** [comment]
here is video of resolution settings
https://youtu.be/ZbngXfIMArQ

**Comment by ToanHoangHuy** [comment]
> here is video of resolution settings https://youtu.be/ZbngXfIMArQ I've deleted dota, cuz its unplayable with that resolution, but fog issue gone

I think that i need to delete and reinstall game again, thanks for your help

**Comment by ToanHoangHuy** [comment]
> here is video of resolution settings https://youtu.be/ZbngXfIMArQ I've deleted dota, cuz its unplayable with that resolution, but fog issue gone

It's work, the graphic is very bad but as long as i can play. Lol, thank you very much.

**Comment by serizawan** [comment]
That tricks work for me as well. Thx @luong-komorebi for pointing me this thread on my other ticket!

**Comment by ToanHoangHuy** [comment]
Yes, the game has got no problems with the trick setting to the lowest resolution. But it still hard to play game, causing difficulty in seeing everything on the map.

**Comment by andreconghau** [comment]
> Try to change resolution, test all kind of resolutions in game This problem was solved for me when I selected the screen resolution 1024x640 (60hz)

Nice! it works fine for me
Thank bro~

**Comment by pototsky-rem** [comment]
nah
This strange behavior of fog of war has also been noticed by Windows users, similar discussions can be found on steam forums. This bug is not related to the OS or chip architecture. I reinstalled Ventura and the fog of war problem went away as if it never existed. I'm using Mac intel chip and I think dota2 is pretty buggy it self

**Comment by brunobertechini** [comment]
It is still happening on Mac M1 (ARM chip). I have the same problem on my M1

**Comment by judedaryl** [comment]
bumping this on my intel macbook 2020, i have an external GPU setup (amd) but when i  upgraded my OS to sonoma yesterday light/fog started bugging out.

**Comment by pototsky-rem** [comment]
> > Try to change resolution, test all kind of resolutions in game This problem was solved for me when I selected the screen resolution 1024x640 (60hz)
> 
> Nice! it works fine for me Thank bro~

This was one of the ideas I came up with to solve this problem, just change the resolution. However, let's be honest, a resolution of 1024x640 (60hz) can hardly be called playable. I also found one solution that is much better and more pleasant, and the problem in the fog of war also disappears. I also made a video https://youtu.be/0tlQtsLyE5g
resolution is set as 2048 x 1280 (60 Hz)

**Comment by JSONBourne** [comment]
Last post from @pototsky-rem solved my issue and the resolution is good too. Make sure you also select the Display Mode to "Exclusive Fullscreen" to see that resolution and then click apply.

**Comment by zzvvmm** [comment]
I found the way to fix it
1. Download the latest MoltenVK lib (currently `[v1.3.0-rc1](https://github.com/KhronosGroup/MoltenVK/releases/tag/v1.3.0-rc1)`) at the below link and extract it somewhere.
https://github.com/KhronosGroup/MoltenVK/releases/
(Remember to download the `MoltenVK-macos.tar`)
2. Copy `libMoltenVK.dylib` to `~/Library/Application\ Support/Steam/steamapps/common/dota\ 2\ beta/game/bin/osx64/` (or wherever you changed the SteamLibrary to) and override the current one.
3. Start game (In my machine it can only start with `sudo`)
```bash
sudo ~/Library/Application\ Support/Steam/steamapps/common/dota\ 2\ beta/game/dota.sh
```

**Comment by dborisoff93-afk** [comment]
+1 on MacOS Tahoe 
Lowering the resolution fixes it but not ideal.
I hope valve works on it. I have to stop playing after so many years. Not necessarily a bad thing though.

Any new solutions guys?
